src/libsync/ocsprofileconnector.cpp

```diff
@@ -0,0 +1,148 @@
+#include "ocsprofileconnector.h"
+#include "common/result.h"
+#include "networkjobs.h"
+#include "iconjob.h"
+#include "theme.h"
+
+#include <QJsonObject>
+#include <QJsonDocument>
+#include <QJsonArray>
+#include <QLoggingCategory>
+#include <QIcon>
+#include <QPainter>
+#include <QImage>
+#include <QSvgRenderer>
+#include <qpixmap.h>
+
+namespace {
+Q_LOGGING_CATEGORY(lcOcsProfileConnector, "nextcloud.gui.ocsprofileconnector", QtInfoMsg)
+
+OCC::HovercardAction jsonToAction(const QJsonObject &jsonActionObject)
+{
+    return {jsonActionObject.value("title").toString("No title"), jsonActionObject.value("icon").toString("no-icon"),
+        jsonActionObject.value("hyperlink").toString("no-link")};
+}
+
+OCC::Hovercard jsonToHovercard(const QJsonArray &jsonDataArray)
+{
+    OCC::Hovercard hovercard;
+    hovercard._actions.reserve(jsonDataArray.size());
+    for (const auto &jsonEntry : jsonDataArray) {
+        Q_ASSERT(jsonEntry.isObject());
+        if (!jsonEntry.isObject()) {
+            continue;
+        }
+        hovercard._actions.push_back(jsonToAction(jsonEntry.toObject()));
+    }
+    return hovercard;
+}
+
+OCC::Optional<QPixmap> createPixmapFromSvgData(const QByteArray &iconData)
```

Comment:
@FlexW Should this really be here? Or, is it better to move it somewhere else (e.g. `IconUtils`) for future use?

Reply:
yes, would make sense

Reply:
As discussed, we leave that for later.